dev-infra/release/publish/actions/tag-recent-major-as-latest.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as semver from 'semver';
+
+import {ReleaseConfig} from '../../config';
+import {ActiveReleaseTrains} from '../../versioning/active-release-trains';
+import {fetchProjectNpmPackageInfo} from '../../versioning/npm-registry';
+import {ReleaseAction} from '../actions';
+import {invokeSetNpmDistCommand, invokeYarnInstallCommand} from '../external-commands';
+
+/**
+ * Release action that tags the recently published major as latest within the NPM
+ * registry. Major versions are published to the `next` NPM dist tag initially and
+ * can be re-tagged to the `latest` NPM dist tag. This allows caretakers to make major
+ * releases available at the same time. e.g. Framework, Tooling and Components
+ * are able to publish v12 to `@latest` at the same time. This wouldn't be possible if
+ * we directly publish to `@latest` because Tooling and Components needs to wait
+ * for the major framework release to be available on NPM.
+ * @see {CutStableAction#perform} for more details.
+ */
+export class TagRecentMajorAsLatest extends ReleaseAction {
+  async getDescription() {
+    return `Tag recently published major v${this.active.latest.version} as "next" in NPM.`;
+  }
+
+  async perform() {
+    await this.checkoutUpstreamBranch(this.active.latest.branchName);
+    await invokeYarnInstallCommand(this.projectDir);
+    await invokeSetNpmDistCommand('latest', this.active.latest.version);
+  }
+
+  static async isActive({latest}: ActiveReleaseTrains, config: ReleaseConfig) {
+    // If the latest release-train does currently not have a major version as
+    // version, this action never should become active.
```

Comment:
The wording is a little weird here.
```suggestion
    // If the latest release-train is already the initial major release (i.e. `X.0.0`) this action
    // should not be active.
```

Reply:
Updated. Let me know if that is better.

<sup>I think the proposed comment isn't expressing what I'm trying to say here; it's the other way around IIUC</sup>